src/uct/tcp/tcp_ep.c

```diff
@@ -546,9 +591,6 @@ static void uct_tcp_ep_handle_disconnected(uct_tcp_ep_t *ep, ucs_status_t status
             ep->flags &= ~UCT_TCP_EP_FLAG_PUT_RX_SENDING_ACK;
         }
 
-        uct_tcp_ep_mod_events(ep, 0, ep->events);
-        ucs_close_fd(&ep->fd);
```

Comment:
what if we call this not from set_ep_failed flow?
keep this as-is, but don't close the fd during uct_tcp_ep_set_failed

Reply:
it is not called from set_failed
it calls `uct_tcp_ep_set_failed()` that closes fd